packages/api-bindings/src/graphql/ReferencePolicy.ts

```diff
@@ -0,0 +1,26 @@
+import { ReferencePolicyType } from '@lens-protocol/domain/src/use-cases/publications';
+
+type FollowersOnlyPolicy = {
+  type: ReferencePolicyType.FOLLOWERS_ONLY;
+};
+
+type UnknownPolicy = {
+  type: ReferencePolicyType.UNKNOWN;
+  contractAddress: string;
+  data: string;
+};
+
+type DegreesOfSeparationPolicy = {
+  type: ReferencePolicyType.DEGREES_OF_SEPARATION;
+  degreesOfSeparation: number;
+};
```

Comment:
The reason I am omitting:
```ts
    commentsRestricted: boolean;
    mirrorsRestricted: boolean;
```

from the "read path" is because it introduces another way of checking for capability of mirroring and/or commenting.

The canonical way of doing that is to use the `canMirror` and `canComment` edges on `Post` and `Comment` nodes:
<img width="284" alt="image" src="https://user-images.githubusercontent.com/3353250/216001529-8a8ddaf7-8034-4e05-b359-ca227f825ae9.png">

Also compared to initial approach where the `ReferencePolicy` (used by the `CreatePostRequest` and `CreateCommentRequest` request models, now called `ReferencePolicyConfig` to make a distinction) I am also flattening out the data (no `params` wrapper). The `params` wrapper is reminiscent of reference module implementation details, instead the `ReferencePolicy` is an abstraction that protect the consumer from such details. Take for example `AnyonePolicy` which is used in case of no-reference module being set up at all for a given publication.

Reply:
Makes sense, good to maintain a single source of truth as to where to check if a publication can be mirrored / commented on 👍